packages/flutter_tools/lib/src/android/build_validation.dart

```diff
@@ -17,6 +17,12 @@ void validateBuild(AndroidBuildInfo androidBuildInfo) {
       'For more information see $kSupportedAbis .'
     );
   }
+  if (!buildInfo.mode.isPrecompiled && buildInfo.analyzeAotSize) {
+    throwToolExit(
+      'AOT size analysis can only be done on AOT builds such as '
+      'profile or release'
+    );
+  }
   if (buildInfo.buildNumber != null) {
```

Comment:
We can also add another check here to crash if analyze is turned on and more than 1 architecture is selected.